core/trino-server-rpm/src/main/rpm/postinstall

```diff
@@ -10,11 +10,14 @@ install --directory --mode=755 /var/log/trino
 # Populate node.id from uuidgen by replacing template with the node uuid
 sed -i "s/\$(uuid-generated-nodeid)/$(uuidgen)/g" /etc/trino/node.properties
 
-# move the trino_env.sh created during pre-install to Trino config location
-if [ -e /tmp/trino_env.sh ]; then
-    mv /tmp/trino_env.sh /etc/trino/env.sh
-else
-    echo "#JAVA_HOME=" >/etc/trino/env.sh
+# read trino_env.sh created during pre-install and save JAVA_HOME in env.sh at Trino config location
+if [ -r /tmp/trino_env.sh ]; then
+    target=/etc/trino/env.sh
+    source /tmp/trino_env.sh
+    sed -i "/^#JAVA_HOME=$/d" $target
+    if ! grep -q '^JAVA_HOME=' $target >/dev/null; then
+        echo "JAVA_HOME=$JAVA_HOME" >> $target
```

Comment:
That's correct, but it's counter intuitive. I just realize we do this to pass detected JAVA_HOME _only_.

thoughts

- instead of genericly named /tmp/trino_env.sh (which we ignore, except for JAVA_HOME), we could have a /tmp/trino-rpm-install-java-home file with just JAVA_HOME _value_. That would reveal its meaning, and avoid false assumptions that whatever was stored in /tmp/trino_env.sh will be propagated to actual env.sh
- the /etc/trino/env.sh can be considered a template, where `#JAVA_HOME=` line gets replaced with `JAVA_HOME=......`, but ordering stays intact. That would allow adding comment above JAVA_HOME, and also avoid problem if someone removes final EOL from the file in the repo, which would cause default java to be used _silently_.

Reply:
I tried replacing instead of appending but I had _a lot_ of issues with escaping slashes in the path to be used with `sed`. The path can contain other special chars than just `/`. I tried changing sed's separator but that didn't work. Can we improve this in a future PR?

Reply:
Done!